app/models/account/provider_methods.rb

```diff
@@ -68,7 +68,7 @@ def build_kind(kind:, **attributes)
       end
     end
 
-    has_many :buyer_accounts, class_name: 'Account', foreign_key: 'provider_account_id', dependent: :destroy, inverse_of: :provider_account do
+    has_many :buyer_accounts, ->(this_account) { where.not(id: this_account.id) }, class_name: 'Account', foreign_key: 'provider_account_id', dependent: :destroy, inverse_of: :provider_account do
```

Comment:
It is caused by the fact that the master account is considered its own provider :upside_down_face: 

You didn't find anything in staging, because in staging/production master's account ID is `6`.
So if you search by ID 6 you will surely find it, the master account's `id` and `provider_account_id` are both equal to `6`.

Reply:
Yes, but my query filters by the `provider` column, not `id`. The query doesn't return master because its `provider` column is `0`.

How was this caused by master being considered its own provider? Could you briefly explain, please?

Reply:
@jlledom , run:
```
select * from accounts where provider_account_id = id
```

master account doesn't seem to be marked as a provider

Reply:
This is where the scope is defined:
https://github.com/3scale/porta/blob/master/app/models/finance/billing_strategy.rb#L46

It doesn't seem to filter based on buyer/provider columns. Maybe that's an inefficiency.

Reply:
The buyers are filtered out later in `BillingWorker::enqueue` https://github.com/3scale/porta/blob/master/app/workers/billing_worker.rb#L39

But yeah, it may make sense to filter them out earlier if we know that buyer accounts will not have buyers of their own anyway.

Reply:
> provider